### ISSUE TITLE
Rename the `_ExpectTypeOf` type

### ISSUE THREAD

**Issue by aryaemami59** [mmkal/expect-type#106]
If `_ExpectTypeOf` is going to be exported, we should probably rename it to something else to avoid confusion between `_ExpectTypeOf` and `ExpectTypeOf`.

**Comment by mmkal** [comment]
Yeah, I don't remember the thinking behind `PositiveExpectTypeOf` and `NegativeExpectTypeOf` and `ExpectTypeOf` being named that way - they're not types for the `expectTypeOf` function, they're asserter-things. Maybe renaming them is a better option. We should check if vitest/anything else publicly on GitHub is using the current names though.

**Comment by aryaemami59** [comment]
@mmkal Vitest is using `ExpectTypeOf` [here](https://github.com/vitest-dev/vitest/blob/8ac70111706a4444a779a75bed34c18b800ea0b3/packages/vitest/src/typecheck/expectTypeOf.ts), I still think we should rename that one too, what do you think?

**Comment by mmkal** [comment]
Hmm vitest is *exporting* it which means god knows how many people are using it!

**Comment by aryaemami59** [comment]
Yeah that's unfortunate.

**Comment by mmkal** [comment]
I'm getting the heebie-jeebies. While the naming is not totally appropriate, it's basically consistent right now. I don't want to make it inconsistent.

I think the most I'd be comfortable with pre-v1 would be `_ExpectTypeOf` -> `CreateExpectTypeOf` or something. Not sure it's even worth it though to be honest. Maybe this should just be a v2 thing.
